### DISCUSSION TITLE
Can't debug actions/loaders in minimal TypeScript+Express app (VS Code)

### DISCUSSION THREAD

**Discussion by rohanliston** [remix-run/remix#4109] [Q&A]
I'd like to integrate Remix into an existing Express app that uses TypeScript. My approach is to generate a fresh Remix app, then incrementally add my server files to ensure it all works as I go along.

I generated a "Just the basics" Remix app and chose Express and TypeScript. I'm able to use the debugger inside loaders at this point.

As soon as I convert `server.js` (the Express app that loads Remix as middleware) to TypeScript (i.e. `server.ts`), I can no longer debug loader functions. Breakpoints do not bind, and `debugger;` statements take me to the generated JavaScript code rather than the original TypeScript source.

What appears to change is that `ts-node` is used instead of `node` for executing the Express server. That's fine (my existing server already uses `ts-node` and the debugger works), but there appears to be something preventing VS Code from seeing the source maps (which are generated, and the paths are all correct).

I've been pulling my hair out for two days on this; can anyone point me in the right direction?

To summarise:
1. Generate "Just the basics" app, choose TypeScript and Express
2. Convert `server.js` to `server.ts` (change `require` to `import`)
3. Add a loader function and insert a breakpoint - it won't bind.
